### PR TITLE
Enable Redis connection pool if commons-pool2 is on the classpath

### PR DESCRIPTION
**Pull request by weixsun**
See #26183  for some more background. jedis turns on the pool by default but lettuce is not.

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java [hunk 1]
```diff
@@ -69,13 +70,29 @@ DefaultClientResources lettuceClientResources() {
 		return DefaultClientResources.create();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean(LettuceClientConfigurationBuilder.class)
+	@ConditionalOnProperty(prefix = "spring.redis.lettuce.pool", value = "enabled", havingValue = "false",
+			matchIfMissing = true)
+	LettuceClientConfigurationBuilder lettuceClientConfigurationBuilder() {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java:77]: Rather then exposing `LettuceClientConfigurationBuilder` as a bean, we should rather apply a `LettuceClientConfigurationBuilderCustomizer` to it when the pool library is on the classpath (and that new `enabled` property is not `false`).

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisProperties.java [hunk 2]
```diff
@@ -421,7 +421,7 @@ public void setPool(Pool pool) {
 		/**
 		 * Lettuce pool configuration.
 		 */
-		private Pool pool;
+		private final Pool pool = new Pool();
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisProperties.java:424]: Jedis should be updated as well I think.

**Review by snicoll — COMMENTED** [review]
@weixsun thanks for the suggestion and the PR! I think Jedis is impacted the same way and pooling should be set automatically if `commons-pool2` is on the classpath.  Would you have time to look at that? If you do, please do not open another PR. This PR can be amended by pushing more commits on your `pool-redis` branch. If you don't have the time, no worries, we can review as part of merging this one.

**Review by weixsun — COMMENTED** [review]
recode.

**Comment by snicoll** [comment]
@weixsun there is no need to rebase your PR against `main` as we will do it as part of reviewing the PR. We prefer to rebase and squash all your commits into one anyway. Please note that every commit you push sends a notification to all watchers of this repository.

**Comment by snicoll** [comment]
Duplicate of #26326

**Pull request closed without merging**